click/core.py

```diff
@@ -208,6 +208,9 @@ class Context(object):
                   codes are used in texts that Click prints which is by
                   default not the case.  This for instance would affect
                   help output.
+    :param show_default: if True, shows defaults for all options.
+                    Even if an option is later created with show_default=False,
+                    this command-level setting overrides it.
```

Comment:
I don't think this should unconditionally override the per-option parameter, unless that's how other context args are working now.

Reply:
But the option's show_default is False by default, so this global command argument does not make sense then. Because that's what it does - shows all defaults without the need to set it explicitly for each option. I can simply drop the last sentence here because it confuses things, in my opinion.

P.S. I did not include any documentation to the .rst files in this PR, do you think it's necessary?

Reply:
Ah, ok, that's fine for now then.